azure/style.yml

```diff
@@ -1,22 +1,31 @@
 parameters:
   setup: []
+  smoketests: true
 
 jobs:
   - template: rustfmt.yml
     parameters:
       name: rustfmt
-  - template: rustfmt.yml
-    parameters:
-      name: rustfmt_beta
-      rust: beta
-      allow_fail: true
+  - ${{ if eq(parameters.smoketests, 'true') }}:
+    - template: rustfmt.yml
+      parameters:
+        name: rustfmt_beta
+        rust: beta
+        allow_fail: true
   - template: cargo-clippy.yml
     parameters:
       name: clippy
       setup: ${{ parameters.setup }}
-  - template: cargo-clippy.yml
-    parameters:
-      name: clippy_beta
-      setup: ${{ parameters.setup }}
-      rust: beta
-      allow_fail: true
+  - ${{ if eq(parameters.smoketests, 'true') }}:
+    - template: cargo-clippy.yml
+      parameters:
+        name: clippy_nowarnings
+        fail_all: true
```

Comment:
This is confusing.  `clippy_nowarnings` but I thought `fail_all` implies warnings?

Reply:
`nowarnings` here is to imply that "no warnings are allowed". I agree this is misleading.

Reply:
I love how we keep reading the opposite thing in each others naming :)